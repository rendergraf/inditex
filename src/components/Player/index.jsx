import React from 'react';
import styled from 'styled-components';

const WrapperPlayer = styled.div`
    position: relative;
    display: flex;
    flex-direction: column;
    background-color: #fff;
    padding: 15px;
    margin: 10px 0;
    box-shadow: 2px 2px 2px rgb(0, 0, 0, 0.15), -1px -1px 1px rgb(0, 0, 0, 0.05);
    background-image: url(${(props) => props.bg});
    background-size: contain;
    background-position: center;
    min-height: 300px;
    height: fit-content;
    background-origin: content-box;
    h1 {
        font-weight: 600;
        padding-bottom: 15px;
        z-index: 9;
        line-height: 1.5rem;
    }
    p {
        padding: 15px 0 80px;
        z-index: 9;
        line-height: 1.5rem;
    }
    &::before {
        content: '';
        position: absolute;
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        background-color: rgba(255, 255, 255, 0.95);
    }
`;

const AudioPlayer = styled.audio`
    width: 100%;
    margin-top: auto;
    z-index: 9;
`;

const Player = ({ episode }) => {
    return (
        <WrapperPlayer bg={episode.artworkUrl600}>
            <h1>{episode.trackName}</h1>
            <p
                dangerouslySetInnerHTML={{
                    __html: episode.description,
                }}
            />
            <AudioPlayer controls name='media'>
                <source src={episode.episodeUrl} type='audio/mpeg' />
            </AudioPlayer>
        </WrapperPlayer>
    );
};

export default Player;
