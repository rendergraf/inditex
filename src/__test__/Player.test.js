import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// Load components
import Player from '../components/Player';

Enzyme.configure({ adapter: new Adapter() });

describe('Player', () => {
    const episode = {
        trackName: 'Test Episode',
        description: 'This is a test episode description',
        episodeUrl:
            'https://traffic.libsyn.com/secure/jbpod/Joe_Budden_Podcast_603.mp3?dest-id=2422538',
        artworkUrl600: 'https://picsum.photos/200/300',
    };
    it('renders correctly', () => {
        const wrapper = shallow(<Player episode={episode} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('displays the episode name', () => {
        const wrapper = shallow(<Player episode={episode} />);
        expect(wrapper.find('h1').text()).toEqual('Test Episode');
    });

    it('displays the episode description', () => {
        const wrapper = shallow(<Player episode={episode} />);
        expect(wrapper.find('p').prop('dangerouslySetInnerHTML'));
    });

    it('displays the audio player with the correct source', () => {
        const wrapper = shallow(<Player episode={episode} />);
        expect(wrapper.find('source').prop('src')).toEqual(
            'https://traffic.libsyn.com/secure/jbpod/Joe_Budden_Podcast_603.mp3?dest-id=2422538'
        );
        expect(wrapper.find('source').prop('type')).toEqual('audio/mpeg');
    });
});
