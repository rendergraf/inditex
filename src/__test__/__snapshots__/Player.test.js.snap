// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player renders correctly 1`] = `ShallowWrapper {}`;
